components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/execution/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -369,7 +370,11 @@ private OperationExecutionResult addToOtherClaims(PerformableOperation operation
         ObjectMapper objectMapper = new ObjectMapper();
         try {
             AccessToken.Claim claim = objectMapper.convertValue(claimToAdd, AccessToken.Claim.class);
-            if (requestAccessToken.getClaim(claim.getName()) != null) {
+            if (SCOPE.equalsIgnoreCase(claim.getName())){
+                return new OperationExecutionResult(operation, OperationExecutionResult.Status.FAILURE,
+                        "Invalid operation path for the scope.");
```

Comment:
Let's mention from where scopes are expected to be modified as well

Reply:
Fixed with 85ffcfd5c4e6f291873513d56309776ccc524748